examples/systemcalls/systemcalls.c

```diff
@@ -93,6 +157,72 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   
 */
 
+    int ret_status = 0;
+    int status;
+
+    pid_t pid;
+    pid = fork();
+ 
+    if(pid == -1)
+    {
+	//Since, new process creation failed, the program is still
+	//in parent context
+	perror("fork");
+	return false;
+    }
+    else if(pid == 0)
+    {
+	//Indicates that the child has been created succesfully
+	//Now, we can proceed ahead to replace the address space of
+	//the process for executing command
+	//printf("I am the child, my pid is %d\n",pid);
+	
+	int file_des = open(outputfile,O_WRONLY | O_CREAT, 0644);
```

Comment:
@Rishab-Shah think about how you could combine logic and avoid duplication with the function above.